pymbar/mbar_solvers/__init__.py

```diff
@@ -0,0 +1,117 @@
+##############################################################################
+# pymbar: A Python Library for MBAR
+#
+# Copyright 2017-2022 University of Colorado Boulder
+# Copyright 2010-2017 Memorial Sloan-Kettering Cancer Center
+# Portions of this software are Copyright (c) 2010-2016 University of Virginia
+# Portions of this software are Copyright (c) 2006-2007 The Regents of the University of California.  All Rights Reserved.
+# Portions of this software are Copyright (c) 2007-2008 Stanford University and Columbia University.
+#
+# Authors: Michael Shirts, John Chodera
+# Contributors: Kyle Beauchamp, Levi Naden
+#
+# pymbar is free software: you can redistribute it and/or modify
+# it under the terms of the MIT License as
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# MIT License for more details.
+#
+# You should have received a copy of the MIT License along with pymbar.
+##############################################################################
+
+"""
+###########
+pymbar.mbar_solvers
+###########
+
+A module implementing the solvers array operations for the MBAR solvers with various code bases for acceleration.
+
+All methods have the same calls and returns, independent of their underlying codes for solution.
+
+Please reference the following if you use this code in your research:
+
+[1] Shirts MR and Chodera JD. Statistically optimal analysis of samples from multiple equilibrium states.
+J. Chem. Phys. 129:124105, 2008.  http://dx.doi.org/10.1063/1.2978177
+
+"""
+
+import logging
+from typing import Union
+
+from .mbar_solver import (
+    validate_inputs,
+    JAX_SOLVER_PROTOCOL,
+    DEFAULT_SOLVER_PROTOCOL,
+    ROBUST_SOLVER_PROTOCOL,
+    BOOTSTRAP_SOLVER_PROTOCOL,
+)
+from .mbar_solver import MBARSolver
+from .numpy_solver import MBARSolverNumpy
+
+logger = logging.getLogger(__name__)
+
+INSTANCED_ACCELERATORS = {}  # Cache the accelerators to avoid re-jit on instancing
+ACCELERATOR_MAP = {"numpy": MBARSolverNumpy}
+default_solver = "numpy"  # Set fallback solver
+
+try:
+    from .jax_solver import MBARSolverJAX
+
+    ACCELERATOR_MAP["jax"] = MBARSolverJAX
+    default_solver = "jax"
+    logger.info("JAX detected. Using JAX acceleration by default.")
+except ImportError:
+    logger.warning(
+        "\n"
+        "********* JAX NOT FOUND *********\n"
+        " PyMBAR can run faster with JAX  \n"
+        " But will work fine without it   \n"
+        "Either install with pip or conda:\n"
+        "      pip install pybar[jax]     \n"
+        "               OR                \n"
+        "      conda install pymbar       \n"
```

Comment:
Should add a `-c conda-forge` in here, I didn't want to use the web UI suggestion features since I wouldn't be able to maintain the formatting of this warning